lib/prometheus/client/histogram.rb

```diff
@@ -94,6 +94,23 @@ def values
         end
       end
 
+      def init_label_set(labels)
+        base_label_set = label_set_for(labels)
+
+        @store.synchronize do
+          [*@buckets, "+Inf"].each do |bucket|
+            # This is basically faster than doing `.merge`
+            bucket_label_set = base_label_set.dup
+            bucket_label_set[:le] = bucket.to_s
+            sum_label_set = base_label_set.dup
+            sum_label_set[:le] = "sum"
+
+            @store.set(labels: bucket_label_set, val: 0)
+            @store.set(labels: sum_label_set, val: 0)
```

Comment:
I found this code a bit hard to read, because you're setting `sum` for each of the buckets in the array.
I think it'd be better if we did `[*@buckets, "+Inf", "sum"]`, instead of `[*@buckets, "+Inf"]`. I may be missing something, but that seems like it'd do it?

Also, i'd use `base_label_set.merge(le: bucket)` instead of the "dup" + setting the key.
It's way more readable, and here we don't care about the tiny performance difference like we do when observing.

Reply:
Not sure is that comment made much sense, sorry :)
I'm proposing this:

```
@store.synchronize do
  (buckets + ["+Inf", "sum"]).each do |bucket|
    @store.set(labels: base_label_set.merge(le: bucket.to_s), val: 0)
  end
end
```

Would that work? Does the test still pass with the same result?

Reply:
> you're setting sum for each of the buckets in the array.

Yeah, there is no need for this! And I agree your suggestions is a lot more readable.